sale_coupon_order_pending/models/coupon_coupon.py

```diff
@@ -18,7 +18,9 @@ def _compute_can_be_applied_to_order(self):
             return
         order = self.env["sale.order"].browse(self.env.context.get("active_id"))
         for coupon in self:
-            if not coupon._check_coupon_code(order):
+            if not coupon._check_coupon_code(
+                order.date_order.date(), coupon.partner_id.id
```

Comment:
You should haved checked how the method is used in this version
```suggestion
                order.date_order.date(), order.partner_id.id, order=order
```